### ISSUE TITLE
IPv6 subnet assignment fails for guest network

### ISSUE THREAD

**Issue by fermosan** [apache/cloudstack#7926]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Networking IP Address provision 
~~~

##### CLOUDSTACK VERSION

~~~
4.18
~~~

##### CONFIGURATION

Advanced Zone
VXLAN isolation
Isolated Network with IPv6 Support

##### OS / ENVIRONMENT

UBUNTU 22.04.3

##### SUMMARY

When applying a dual stack networking offering to an existing or new network the guest ipv6 network is not properly assigned.
The UI in the Details of the Guest Network is indicating "undefined via 2a0f:ffc0:......".  
The Virtual Router has no IPv6 address assigned on the guest network. The public IPv6 address is applied and working as expected. 
The Management server log  doesn't report any relevant errors. 
##### STEPS TO REPRODUCE

Create a new Zone with VXLAN isolation method
Add host
Define IPv4 / IPv6 guest/public networks
Create a dual stack network offering
Create a new guest network 
Create a new VM Instance, assign it to the network and start it

Setup of the PUBLIC network in the Zone
![Screenshot 2023-08-31 015017](https://github.com/apache/cloudstack/assets/78906279/605d8c01-eb6e-4d95-9e09-138d534d67f9)

Setup of the Guest network in the Zone
![Screenshot 2023-08-31 015910](https://github.com/apache/cloudstack/assets/78906279/02ddd256-a27f-440c-8653-7ee3fe13aadf)

Status of Guest Network
![Screenshot 2023-08-31 020033](https://github.com/apache/cloudstack/assets/78906279/1a9afe7e-f4a0-430d-ba09-3e32847a0a96)

~~~

~~~

##### EXPECTED RESULTS

~~~
The Guest network should be assigned a /64 IPv6 Prefix
~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by weizhouapache** [comment]
@shwstppr 
can you please have a look ? thanks

**Comment by shwstppr** [comment]
@weizhouapache could be something to do with VXLAN isolation because IPv6 smoke tests seem to be running okay and manual deployment also works. Need to check
cc @fermosan

**Comment by weizhouapache** [comment]
@shwstppr thanks for the reply.
yes, could be.
the guest ipv6 range should be allocated, no matter what isolation method is used.

**Comment by fermosan** [comment]
@shwstppr @weizhouapache A similar setup with VLAN isolation is working as expected. I am not sure if the isolation type makes any noticeable difference.  
Also, if this is something easy to fix can you please include it in the upcoming 4.18.1 since it's kind of critical for our operation.

**Comment by fermosan** [comment]
Some more debugging info i just collected (taking a look at database side atm). 
The ip6_guest_prefix_subnet_network_map table has the correct entry for the allocation pointing to network id 205.
-----------------------------
id: 2
uuid: a09f1e7d-b64b-441c-a36d-e0abd5068cff
prefix_id: 2
subnet: 2a0f:xxxx:xxxx:xxxx::/64
network_id: 205
state: Allocated
updated: 2023-08-30 22:19:42
created: 2023-08-30 22:19:42
removed: NULL
-----------------------------

Network with ID 205 has no ip6_gateway and ip6_cidr populated. 
-----------------------------
id: 205
name: dualstack network
uuid: bdfbb100-b9fb-47a3-8b0f-abbb42cb1781
display_text: dualstack network
traffic_type: Guest
broadcast_domain_type: Vxlan
broadcast_uri: vxlan://103744
gateway: 10.1.1.1
cidr: 10.1.1.0/24
mode: Dhcp
network_offering_id: 22
physical_network_id: 200
data_center_id: 1
guru_name: VxlanGuestNetworkGuru
state: Implemented
related: 205
domain_id: 1
account_id: 2
dns1: NULL
dns2: NULL
ip6dns1: NULL
ip6dns2: NULL
guru_data: NULL
set_fields: 0
acl_type: Account
network_domain: cs2cloud.internal
reservation_id: 831389b3-cce4-475c-a7a2-e304280a4c5a
guest_type: Isolated
restart_required: 0
created: 2023-08-30 22:19:42
removed: NULL
specify_ip_ranges: 0
vpc_id: NULL
ip6_gateway: NULL
ip6_cidr: NULL
network_cidr: NULL
display_network: 1
network_acl_id: NULL
streched_l2: 0
redundant: 0
external_id: NULL
public_mtu: 1500
private_mtu: 1500
-----------------------------


@shwstppr

**Comment by weizhouapache** [comment]
thanks for the info @fermosan 

@shwstppr has made some changes for ipv6 in ExternalGuestNetworkGuru

https://github.com/apache/cloudstack/blob/78411fd405c6dfe83381f010c143d6bb90b6b7e4/server/src/main/java/com/cloud/network/guru/ExternalGuestNetworkGuru.java#L130-L140

it looks we need to make similar changes in VxlanGuestNetworkGuru

cc @shwstppr

**Comment by shwstppr** [comment]
Thanks for the pointer @weizhouapache 
Will try to test the change tomorrow

**Comment by fermosan** [comment]
@shwstppr Great. Tyvm. 

@weizhouapache If there is a fix can we include it in 4.18.1.0 ?

**Comment by soreana** [comment]
@weizhouapache I put this one in the 4.18.1 milestone as its pull request also targeted for that milestone.

**Comment by weizhouapache** [comment]
> @weizhouapache I put this one in the 4.18.1 milestone as its pull request also targeted for that milestone.

thanks @soreana

**Comment by weizhouapache** [comment]
fixed by #7933
